Marlin/stepper.cpp

```diff
@@ -574,8 +574,11 @@ void set_stepper_direction() {
 // block begins.
 FORCE_INLINE void trapezoid_generator_reset() {
 
-  if (current_block->direction_bits != out_bits) {
+  static unsigned char last_extruder = 0;
```

Comment:
Might as well make it (a) `int8_t` for consistency. And initialize it to `-1` so that even extruder 0 will be initialized on the first move.
```cpp
static int8_t last_extruder = -1;
```